src/main.zig

```diff
@@ -2704,9 +2704,22 @@ pub fn cmdFmt(gpa: *Allocator, args: []const []const u8) !void {
             fatal("cannot use --stdin with positional arguments", .{});
         }
 
-        const stdin = io.getStdIn().reader();
-
-        const source_code = try stdin.readAllAlloc(gpa, max_src_size);
+        const stdin = io.getStdIn();
+
+        const source_code = blk: {
+            const source_code = try stdin.readToEndAllocOptions(gpa, max_src_size, null, @alignOf(u16), null);
+            errdefer gpa.free(source_code);
+
+            // If the file starts with a UTF-16 BOM, translate it to UTF-8
+            if (mem.eql(u8, source_code[0..2], "\xff\xfe")) {
+                const source_code_utf16_le = mem.bytesAsSlice(u16, source_code);
+                const source_code_utf8 = try std.unicode.utf16leToUtf8Alloc(gpa, source_code_utf16_le);
+                gpa.free(source_code);
+                break :blk source_code_utf8;
+            } else {
```

Comment:
Nit: Should we detect the other interesting BOMs (UTF16BE and UTF32LE/BE) and print an error?

Reply:
This was added in 72079e1553bb4a40c927aeec9ce7725e4dc68dbe.

However, as said in the commit message, the BOM of UTF-32 is identical to the BOM of UTF-16 followed by a null character. Therefore, one check is redundant, and an invalid source file in UTF-16 containing a null character at the start will be mistaken for an UTF-32 file, which also results in an error.